tests/data/test_subject.py

```diff
@@ -106,3 +106,24 @@ def test_bad_arg(self):
     def test_no_images(self):
         with self.assertRaises(ValueError):
             tio.Subject(a=0)
+
+    def test_copy_subclass(self):
+        class CustomSubject(tio.Subject):
+            def __init__(self, img: tio.Image, *args, **kwargs):
+                kwargs['img'] = img
+                super().__init__(*args, **kwargs)
+        tensor = torch.rand(1, 3, 3, 4)
+        subject = CustomSubject(img=tio.ScalarImage(tensor=tensor))
+        new_subject = copy.copy(subject)
+        assert isinstance(new_subject, CustomSubject)
+
+    def test_copy_subclass_no_kwargs(self):
+        class CustomSubject(tio.Subject):
+            def __init__(self):
+                tensor = torch.rand(1, 3, 3, 4)
+                img = tio.ScalarImage(tensor=tensor)
+                kwargs = {'img': img}
+                super().__init__(**kwargs)
+        subject = CustomSubject()
+        new_subject = copy.copy(subject)
+        assert isinstance(new_subject, CustomSubject)
```

Comment:
```suggestion
    def test_copy_subclass(self):
        class CustomSubject(tio.Subject):
            pass
        assert isinstance(new_subject, CustomSubject)
```

IMO overything else (including argument forwarding) tests the correct implementation of the subclass which is up to the user